platform/default/default_file_source.cpp

```diff
@@ -237,22 +238,18 @@ void DefaultFileSource::setResourceTransform(std::function<std::string(Resource:
 }
 
 std::unique_ptr<AsyncRequest> DefaultFileSource::request(const Resource& resource, Callback callback) {
-    class DefaultFileRequest : public AsyncRequest {
-    public:
-        DefaultFileRequest(Resource resource_, FileSource::Callback callback_, util::Thread<DefaultFileSource::Impl>& thread_)
-            : thread(thread_),
-              workRequest(thread.invokeWithCallback(&DefaultFileSource::Impl::request, this, resource_, callback_)) {
-        }
+    auto mailbox = std::make_shared<Mailbox>(*util::RunLoop::Get());
+    auto request = std::make_unique<FileSourceRequest>(mailbox);
 
-        ~DefaultFileRequest() override {
-            thread.invoke(&DefaultFileSource::Impl::cancel, this);
-        }
+    request->onResponse(std::move(callback));
+    request->onCancel([fs = impl->actor(), req = request.get()] () mutable { fs.invoke(&Impl::cancel, req); });
 
-        util::Thread<DefaultFileSource::Impl>& thread;
-        std::unique_ptr<AsyncRequest> workRequest;
-    };
+    thread->invoke(&Impl::request, request.get(), resource, [ref =
```

Comment:
Great idea, this is not only more Actor-ish but I could get rid of lambdas altogether on all file sources but the default.